libuuu/sdps.cpp

```diff
@@ -145,6 +157,10 @@ int SDPSCmd::run(CmdCtx *pro)
 	if (m_bskipflashheader)
 		offset += GetFlashHeaderSize(p, offset);
 
+	// Detect barebox binaries that have the IVT header at offset 32K
+	if (!is_ivt_header(p, offset) && is_ivt_header(p, offset + 0x8000))
+		offset += 0x8000;
```

Comment:
```p = p1->request_data(0,  offset+8000+sizeof(ivt));```

Reply:
Need call request_data to fetch data from stroage, such as http download or decompress from bz2/zstd

Reply:
Why? There is already either `p1->request_data(0, m_scan_limited);` or `p1->request_data(0, UINT64_MAX);` a few lines above.

Reply:
At least, it'd better check (offset+0x8000 + sizeof(ivt)) < p->size()

Reply:
This is already done inside `is_ivt_header(p, offset + 0x8000))` or am I missing something?

Reply:
Yes! Thanks